### PR TITLE
IDTCTJNKNS-189: Update or add tests related to new AirGap strategy work

### PR DESCRIPTION
**Pull request by DMaxfield-BDS**

### CODE REVIEW

File: src/main/java/com/synopsys/integration/jenkins/detect/service/strategy/DetectAirGapJarStrategy.java [hunk 1]
```diff
@@ -91,18 +91,18 @@ public SetupCallableImpl(JenkinsIntLogger logger, Map<String, String> environmen
             RemoteJavaService remoteJavaService = new RemoteJavaService(logger, remoteJdkHome, environmentVariables);
             String javaExecutablePath = remoteJavaService.calculateJavaExecutablePath();
 
-            logger.info("Detect configured: " + airGapJarPath);
+            logger.info("Detect AirGap jar configured: " + airGapJarPath);
```
Comment (DMaxfield-BDS) [inline src/main/java/com/synopsys/integration/jenkins/detect/service/strategy/DetectAirGapJarStrategy.java:94]: Changed for clarity, both in the code and for the customer

File: src/main/java/com/synopsys/integration/jenkins/detect/service/strategy/DetectAirGapJarStrategy.java [hunk 2]
```diff
@@ -91,18 +91,18 @@ public SetupCallableImpl(JenkinsIntLogger logger, Map<String, String> environmen
...
             try {
-                airGapInstallation = jenkinsConfigService.getInstallationForNodeAndEnvironment(DetectAirGapInstallation.DescriptorImpl.class, airGapDownloadStrategy.getAirGapInstallationName())
-                                         .orElseThrow(() -> new DetectJenkinsException(String.format(
-                                             "Problem encountered getting Detect Air Gap tool with the name %s from global tool configuration. Check Jenkins plugin and tool configuration.",
-                                             airGapDownloadStrategy.getAirGapInstallationName())));
+                String airGapInstallationName = airGapDownloadStrategy.getAirGapInstallationName();
+                airGapInstallation = jenkinsConfigService.getInstallationForNodeAndEnvironment(DetectAirGapInstallation.DescriptorImpl.class, airGapInstallationName).orElseThrow(
+                    () -> new DetectJenkinsException(
+                        String.format("Problem encountered getting Detect Air Gap tool with the name %s from global tool configuration. Check Jenkins plugin and tool configuration.", airGapInstallationName)));
```
Comment (DMaxfield-BDS) [inline src/main/java/com/synopsys/integration/jenkins/detect/service/strategy/DetectAirGapJarStrategy.java:105]: Broke out variable for clarity.

File: src/test/java/com/synopsys/integration/jenkins/detect/service/strategy/DetectJarStrategyTest.java [hunk 3]
```diff
@@ -63,12 +63,6 @@ public void testArgumentEscaper() {
         assertEquals(Function.identity(), detectJarStrategy.getArgumentEscaper());
     }
 
-    @Test
-    public void testInvocationParameters() {
-        DetectJarStrategy detectJarStrategy = new DetectJarStrategy(logger, environmentVariables, remoteJdkHome, detectJarPath);
-        assertEquals(Arrays.asList(expectedJdkJavaPath, "-jar", detectJarPath), detectJarStrategy.getInitialArguments(expectedJdkJavaPath));
-    }
-
```
Comment (DMaxfield-BDS) [inline src/test/java/com/synopsys/integration/jenkins/detect/service/strategy/DetectJarStrategyTest.java:71]: getInitialArguments() was removed, and logic is part of call()

File: src/main/java/com/synopsys/integration/jenkins/detect/service/strategy/DetectAirGapJarStrategy.java (hunk 1 above)
Comment (rottebds) [reply src/main/java/com/synopsys/integration/jenkins/detect/service/strategy/DetectAirGapJarStrategy.java:94]: 👍

**Review by rottebds — COMMENTED** [review]
Worth noting-- when reviewing these tests I realized that a significant number of them will only pass on linux-based operating systems due to literal string comparisons with file paths. This is not a big deal, but we should consider improving it if we ever intend on a non-linux based developer to build the plugin.

This is also true of DetectJarStrategyTest.

File: src/test/java/com/synopsys/integration/jenkins/detect/service/strategy/DetectAirGapJarStrategyTest.java [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+import com.synopsys.integration.jenkins.service.JenkinsConfigService;
+import com.synopsys.integration.log.LogLevel;
+import com.synopsys.integration.util.IntEnvironmentVariables;
+
+import hudson.model.TaskListener;
+import jenkins.security.MasterToSlaveCallable;
+
+public class DetectAirGapJarStrategyTest {
+    private static final String DETECT_JAR_PREFIX = "synopsys-detect-";
+    private static final String DETECT_JAR_SUFFIX = ".jar";
```
[the remark this answers is not in the document]
Comment (DMaxfield-BDS) [reply src/test/java/com/synopsys/integration/jenkins/detect/service/strategy/DetectAirGapJarStrategyTest.java:39]: I updated the fields to make them public.

I will also open a ticket to update the tests for Windows environments. This is a 100% valid point, and we need to handle this.

**Pull request merged**